templates/webhook.bin.erb

```diff
@@ -352,6 +370,7 @@ class Server < Sinatra::Base
         end
         status_message =  {:status => :success, :message => message.to_s, :branch => branch, :status_code => 200}
         $logger.info("message: #{message} branch: #{branch}")
+        generate_types(branch) if types?
```

Comment:
What happens if the branch is being deleted?

In general, I've always thought it would be good if somewhere we set whether the hook was for a branch create, modify or delete.  I guess doing this would need a bit of a refactor.  The way we set 'branch' is already a little ugly.  Maybe we could extract a whole bunch of things.  It'd be quite nice to be able to include a bit more detail in the slack notifications etc.